### DISCUSSION TITLE
how you start the APIs in Tomcat

### DISCUSSION THREAD

**Discussion by masoudmatak** [keeps/roda#3057] [General]
I have build RODAs war-file roda-wui-5.4.0-SNAPSHOT.war and deployed it in Tomcat 9.0 , when I start  https://locakhost:8080/roda-wui-5.4.0-SNAPSHOT you see just a html-page as a text-info.. nothing to execure!   the same if you try with https://locakhost:8080/roda-wui-5.4.0-SNAPSHOT/api/  nothing will happend, just error page..  Is it the way RODA works?
is there any documentation for deploying RODA locally?

**Comment by luis100** [comment]
Yes, there is: https://github.com/keeps/roda/blob/master/deploys/standalone/README.md

**Comment by masoudmatak** [comment]
Thank you Luis for the replay!
I am using Tomcat for the deployment.
what you mean with Yes?
 Do you mean that if you deploy the application in Tomcat locally, is it just a static html-page?

and what is the entry to check the API:s if they are deployed in Apache locally? give just a sample API-entry


what I see after my Tomcat deployemenmt:

[cid:92f3e4f5-fcf4-46c3-b199-dd1c1b3af112]

and the below figure  is your demo-page:

[cid:919703b7-cc25-4585-9190-b8e2c4ceaf1f]
________________________________
Från: Luis Faria ***@***.***>
Skickat: den 22 januari 2024 17:30
Till: keeps/roda ***@***.***>
Kopia: masoudmatak ***@***.***>; Author ***@***.***>
Ämne: Re: [keeps/roda] how you start the APIs in Tomcat (Discussion #3057)


Yes, there is: https://github.com/keeps/roda/blob/master/deploys/standalone/README.md

**Reply by luis100** [reply]
The way we recommend deploying is by using docker images. RODA requires several dependant services to start up, they are part of the [docker compose](https://github.com/keeps/roda/blob/master/deploys/standalone/docker-compose.yaml) shown. Also, it requires a set of configuration to be included, which is also in the docker compose and docker image. Finally, it requires to be setup in a particular way inside the Tomcat, which again is described in the [Dockerfile](https://github.com/keeps/roda/blob/master/roda-ui/roda-wui/docker-files/Dockerfile).

Deploying this in bare metal, directly in Tomcat, would be a complex setup.

Please take the time to read through the provided link and the documentation linked there.

https://github.com/keeps/roda/blob/master/deploys/standalone/README.md
